File: app/src/main/java/mb00/android/codehub/api/RetrofitRawBuilder.java
package mb00.android.codehub.api;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;


public class RetrofitRawBuilder {

    private static Retrofit retrofitInstance;

    private RetrofitRawBuilder() {}

    public static Retrofit getInstance() {
        if (retrofitInstance == null) {
            retrofitInstance = new Retrofit.Builder()
                    .baseUrl("https://raw.githubusercontent.com/")
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofitInstance;
    }

}

File: app/src/main/java/mb00/android/codehub/api/builder/RetrofitRawBuilder.kt
package mb00.android.codehub.api.builder

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory


object RetrofitRawBuilder {

    private var retrofitInstance: Retrofit? = null

    val instance: Retrofit
        get() {
            if (retrofitInstance == null) {
                retrofitInstance = Retrofit.Builder()
                        .baseUrl("https://raw.githubusercontent.com/")
                        .addConverterFactory(GsonConverterFactory.create())
                        .build()
            }
            return retrofitInstance!!
        }

}
